### ISSUE TITLE
О подписях рисок по бокам виджета

### ISSUE THREAD

**Issue by RandomAmbersky** [DarkHobbit/chronoline#27]
![2016-02-05 20 46 04](https://cloud.githubusercontent.com/assets/436951/12865053/68201bdc-ccb2-11e5-8bb0-9bc0ddf55770.png)

Полагаю, выбор риски справа, которую нужно подписывать, определять не по уходу самой риски за пределы рабочей зоны, а по уходу правого края подписи...  

![2016-02-06 9 12 48](https://cloud.githubusercontent.com/assets/436951/12865055/6e8ce9b4-ccb2-11e5-9a4c-6864fd0b1446.png)

**Comment by DarkHobbit** [comment]
А вот это как раз следствие принятого решения, что фиксированная ширина не у главной единицы, а у дня. Соответственно, количество главных единиц на шкале _априорно_ можно определить только приблизительно, и в примере как раз тот случай, когда эта погрешность вылезла наружу. (Впрочем, в отвергнутом варианте, когда постоянная ширина у основной единицы, проблемы куда более серьёзные, так что поступили мы, имхо, правильно.)
Да, его можно определить апостериорно - расставить метки, но прежде чем делать подпись под N-ной меткой, проверить, можно ли её сделать под N+1-й, не вылезая за шкалу (благо ширину текста мы теперь определять умеем), и если можно, то так и поступать.

**Comment by DarkHobbit** [comment]
Да, и не совсем правильно, наверное, называть ChronoLine плагином - это ведь не DLL и тем более не DLL, которая построена по API, рассчитанному на подключение целого семейства DLL (плагином уместно называть, IMHO, только последний случай, вот в Амбернете были НАСТОЯЩИЕ плагины).
А ChronoLine - это библиотека, скорее...

**Comment by DarkHobbit** [comment]
Тем более в данном случае имелись в виду не бока программного кода, а бока визуального элемента (виджета), так и подправил...

**Comment by DarkHobbit** [comment]
Иногда, кстати, из-за то
